### ISSUE TITLE
Create additional assertions via contexts

### ISSUE THREAD

**Issue by azaroth42** [w3c/json-ld-syntax#64]
(This is a meta-issue)

Many existing JSON structures use simple "short cut" properties when there is only a single piece of information known about an intermediate resource.  For example, if only the name of the author of a book is known then a `creator` property with a literal value is shorthand for a longer assertion chain of that the book was created by some actor that has the given name.

Thus:

```
{
  "type": "Book",
  "creator": "J.R.R. Tolkien"
}
```

Could be expanded to:

```
{
  "type": "Book",
  "created_by": {
    "type": "Actor",
    "label": "J.R.R. Tolkien"
  }
}
```

This could be added to the context for creator:

```
"creator": {
  "@type": "@nest", 
  "@values": {"@type": "Actor", "@property": "label", "@id": "created_by"}
}
```

Or similar.

**Comment by azaroth42** [comment]
I'm a strong :-1: on this sort of thing. I believe that contexts should not do data transformation, only assert a mapping between JSON keys and the graph structure that already exists.  This is a reductio argument against #7 and #15.

**Comment by iherman** [comment]
We do have exactly this issue in the [Web Publication Manifest](https://w3c.github.io/wpub/#wp-properties) and the draft includes a separate ["canonical manifest" section](https://w3c.github.io/wpub/#canonical-manifest) that defines a number of transformation steps like this one.

However, even with this example around, I am wary of such feature for JSON-LD 1.1. We should not use the context feature for data transformation. It may open pandora's box. We may want to look at this issue in general at some point, but that may have to be treated separately and probably needs a longer incubator period. For example, even if we did it, we should may want to have an explicitly different mechanism defined for this, not part of `@context` but a separate `@transform` patterns to make it more explicit and not make `@context` even more complex that it is. It may also touch on RDF graph transformations, ie, not necessarily bound to a particular serialization.

At this moment, I am 👎 on this...

**Comment by BigBlueHat** [comment]
FWIW, Google's [Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool) (which only supports one vocabulary: Schema.org) makes some undeclared assumptions (based on that single vocabulary) which amount to this sort of coercion.

You can see a more complete description in [this comment over at the WPUB repo](https://github.com/w3c/wpub/issues/314#issuecomment-415885945), but essentially, there's an unexpressed "default" `Thing` class into which string values get coerced:

```json
{"@context": "http://schema.org/", "publisher": "asdf", "@type": "Book"}
```
becomes
```json
{"@context": "http://schema.org/", "publisher": {"@type": "Thing", "name": "asdf"}, "@type": "Book"}
```

This does feel like it moves the needle from vocabulary into ontology.

**Comment by azaroth42** [comment]
On the call of 2018-09-07, the WG [resolved](https://www.w3.org/2018/json-ld-wg/Meetings/Minutes/2018/2018-09-07-json-ld#resolution5) to close this issue won't fix, with the same rationale as #7. The use cases should be discussed in the CG.
